### PR TITLE
Issue 24496 generate skeleton thumbnail on broken thumbnail creation proccess

### LINKED ISSUE #24496: Generating a thumbnail for /members/index fails.. Again, we should be able to save a favorite even without a thumbnail.  On the favorites listing, if we don't have a thumbnail, we should show a skeleton.
### Parent task
https://github.com/dotCMS/core/issues/24326

### Description

skeleton thumbnail when no screenshot was created/saved from html
image

skeleton thumbnail when error creating screenshot from html
image

### Steps:
1) Go to a Site Browser, and open `Colorado Preseason Special`
2) try to create a favorite page and a generic thumbnail will be shown
3) go to pages portlet and verify the generic thumbnail is shown

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes

skeleton thumbnail when no screenshot was created/saved from html
image

skeleton thumbnail when error creating screenshot from html
image

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss [hunk 1]
```diff
@@ -1,7 +1,9 @@
 dot-html-to-image {
+    align-items: center;
+    background-color: #fafafa;
```
Comment (hmoreras) [inline core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss:3]: use standard color ?

Comment (alfredo-dotcms) [reply core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss:3]: mmm, this is a dotcms-webcomponents and Will suggested that color. @fmontes thoughts?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.html [hunk 2]
```diff
@@ -8,7 +8,15 @@
 >
     <form class="p-fluid" *ngIf="form" [formGroup]="form" data-testId="form">
         <div class="fields-col">
-            <div class="dot-favorite-dialog__thumbnailField" data-testId="thumbnailField">
+            <dot-pages-favorite-page-empty-skeleton
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.html:11]: if else.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts [hunk 3]
```diff
@@ -160,8 +160,15 @@ export class DotFavoritePageComponent implements OnInit, OnDestroy {
...
+            ? dotHtmlToImageElement.addEventListener('pageThumbnail', (event: CustomEvent) => {
+                  if (event.detail.file) {
+                      this.form.get('thumbnail').setValue(event.detail.file);
+                  } else {
+                      this.form.get('thumbnail').setValue('');
+                      this.store.setShowFavoriteEmptySkeleton(true);
+                  }
+              })
+            : null;
     }
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts:173]:
```suggestion
        if (dotHtmlToImageElement) {
          dotHtmlToImageElement.addEventListener('pageThumbnail', (event: CustomEvent) => {
                  if (event.detail.file) {
                      this.form.get('thumbnail').setValue(event.detail.file);
                  } else {
                      this.form.get('thumbnail').setValue('');
                      this.store.setShowFavoriteEmptySkeleton(true);
                  }
              })
        }
    }
```

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts:173]:
```suggestion
          dotHtmlToImageElement?.addEventListener('pageThumbnail', (event: CustomEvent) => {
                  if (event.detail.file) {
                      this.form.get('thumbnail').setValue(event.detail.file);
                  } else {
                      this.form.get('thumbnail').setValue('');
                      this.store.setShowFavoriteEmptySkeleton(true);
                  }
              })
        }
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-favorite-panel/dot-pages-card/dot-pages-card.component.html [hunk 4]
```diff
@@ -2,11 +2,13 @@
     <ng-template pTemplate="header">
         <div
             class="dot-pages-favorite-card-content__image"
+            *ngIf="imageUri"
             [ngStyle]="{ 'background-image': 'url(' + imageUri + ')' }"
             data-testId="favoriteCardImageContainer"
         >
             <img [alt]="title" [src]="imageUri" />
         </div>
+        <dot-pages-favorite-page-empty-skeleton *ngIf="!imageUri" />
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-favorite-panel/dot-pages-card/dot-pages-card.component.html:11]: if else.

File: core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss (hunk 1 above)
Comment (fmontes) [reply core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss:3]: That's the lightest gray we have in the list, go with vars.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.html (hunk 2 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.html:11]: fixed!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-favorite-panel/dot-pages-card/dot-pages-card.component.html (hunk 4 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-favorite-panel/dot-pages-card/dot-pages-card.component.html]: fixed!

File: core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss (hunk 1 above)
Comment (alfredo-dotcms) [reply core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.scss]: fixed!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts (hunk 3 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts]: fixed!

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**